### ISSUE TITLE
Add Async support

### ISSUE THREAD

**Issue by thargy** [Sergio0694/ComputeSharp#80]
Currently [`ShaderRunner<T>.Run` ](https://github.com/Sergio0694/ComputeSharp/blob/91f94abf87ce6cace7885ab6d68099a925f55d5c/src/ComputeSharp.Shaders/ShaderRunner%7BT%7D.cs#L85) only provides a synchronous blocking method.

I propose that you consider adding a `RunAsync()` overload, and corresponding `ForAsync()` extension methods that are non-blocking, as the call could be 'long running'.  You may wish to change `ShadersCache` to a `ConcurrentDictionary` to avoid the locks.

**Comment by Sergio0694** [comment]
Hey @thargy! Sorry it took a minute, this is a great ideal! 😄
You'll be happy to hear that this feature is now implemented with #184.
It leverages `IAsyncDisposable` through the new `ComputeContext` APIs:

```csharp
await using (var context = GraphicsDevice.Default.CreateComputeContext())
{
    context.For(100, new MyShader(buffer));
}
```

Just use the `ComputeContext` APIs as usual, and add an `await` right before the `using`, and that's it! ComputeSharp will execute all the queued operations (shader dispatches, clears, resource barriers, etc.) asynchronously when the context goes out of scope.

Let me know what you think! Happy new year! 🎉

**Comment by HoneyFox** [comment]
Hi, @Sergio0694, I think what @thargy wanted is something like this:
```
AsyncComputeTask task = null;
using (var context = GraphicsDevice.Default.CreateAsyncComputeContext(out task))
{
    // Long running GPU compute workload.
}
// do other CPU tasks here.
task.Wait();
// copy data from GPU back to CPU here.
```
